trie/keyBuilder/keyBuilder.go

```diff
@@ -0,0 +1,56 @@
+package keyBuilder
+
+import (
+	"github.com/ElrondNetwork/elrond-go/common"
+)
+
+// NibbleSize marks the size of a byte nibble
+const NibbleSize = 4
+
+type keyBuilder struct {
+	key []byte
+}
+
+// NewKeyBuilder creates a new key builder. This is used for building trie keys when traversing the trie.
+// Use this only if you traverse the trie from the root, else hexToTrieKeyBytes might fail
+func NewKeyBuilder() *keyBuilder {
+	return &keyBuilder{
+		key: []byte{},
+	}
+}
+
+// BuildKey appends the given byte array to the existing key
+func (kb *keyBuilder) BuildKey(keyPart []byte) {
+	kb.key = append(kb.key, keyPart...)
+}
+
+// GetKey transforms the key from hex to trie key, and returns it
+func (kb *keyBuilder) GetKey() ([]byte, error) {
+	return hexToTrieKeyBytes(kb.key)
+}
+
+// Clone returns a new KeyBuilder with the same key
+func (kb *keyBuilder) Clone() common.KeyBuilder {
+	return &keyBuilder{
+		key: kb.key,
```

Comment:
Discussed. It works because everything is done in a serial manner, and each trie gets a new keyBuilder. And when GetKey() is called, it copies the data to a new pointer.